test/tests/interfaces/pbs_sched_interface_test.py

```diff
@@ -0,0 +1,143 @@
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.interfaces import *
+from ptl.lib.pbs_ifl_mock import MGR_CMD_CREATE
+
+
+@tags('multisched')
+class TestSchedulerInterface(TestInterfaces):
+    """
+    Test suite to test different scheduler interfaces
+    """
+
+    def setUp(self):
+        # delete all schedulers except default
+        self.server.manager(MGR_CMD_DELETE, SCHED, id="@default")
+        TestInterfaces.setUp(self)
+
+    def test_create_successful_scheduler_object(self):
+        """
+        Creates a scheduler object succcessfully .
+        """
+        a = {'sched_cycle_length': 12000}
+        rc = self.server.manager(MGR_CMD_CREATE, SCHED, a, id="test1")
+        self.assertTrue(rc == 0)
+
+    def test_list_schedulers(self):
+        """
+        List a scheduler object.
+        """
+        pass
+
+    def test_delete_schedulers(self):
+        """
+        Delete a scheduler object succcessfully .
+        """
+        a = {'sched_cycle_length': 12000}
+        rc = self.server.manager(MGR_CMD_CREATE, SCHED, a, id="test1")
+        rc = self.server.manager(MGR_CMD_DELETE, SCHED, a, id="test1")
+        self.assertTrue(rc == 0)
+
+    def test_duplicate_scheduler_name(self):
+        """
+        Check for the scheduler object name.
+        """
+        a = {'sched_cycle_length': 12000}
+        rc = self.server.manager(MGR_CMD_CREATE, SCHED, a, id="test1")
+        try:
+            rc = self.server.manager(MGR_CMD_CREATE, SCHED, a, id="test1")
+        except PbsManagerError, e:
+            self.assertTrue(
+                'qmgr: Error (15211) returned from server' in e.msg[1])
+
+    def test_permission_issue_on_sched(self):
+        """
+        Check for the permission to create/delete/modify scheduler object.
+        """
+        try:
+            self.server.manager(MGR_CMD_CREATE, SCHED,
+                                id="test1",
+                                runas=OPER_USER)
+        except PbsManagerError, e:
+            if self.server.get_op_mode() == PTL_CLI:
+                self.assertTrue(
+                    'qmgr: Error (15007) returned from server' in e.msg[1])
+
+        try:
+            self.server.manager(MGR_CMD_CREATE, SCHED,
+                                id="test2",
+                                runas=ROOT_USER)
+        except PbsManagerError, e:
+            self.fail(e.msg)
+
+    def test_delete_default_sched(self):
+        """
+        Delete default scheduler.
+        """
+        a = {}
+        try:
+            rc = self.server.manager(MGR_CMD_DELETE, SCHED, a, id="main")
+        except PbsManagerError, e:
+            if self.server.get_op_mode() == PTL_CLI:
```

Comment:
No, i am expecting PBS error codes(here  15211, 15212, 15214) to be set in Exception class. So that i can assert the same

rc has different value. Dont know mapping with pbs error code.